### PR TITLE
Add RLE support for crowds

### LINKED ISSUE #14: Handle RLE masks
Now only polygon masks are supported. We need to handle RLE masks as well.

### PR DESCRIPTION
**Pull request by heada**
Fix #14 
I have impelmented support for uncompressed rle

### CODE REVIEW

**Comment by trsvchn** [comment]
@heada Thank you for the PR!
But it looks like this PR reverts 2 previous commits: object based coloring and save file fix
could you please to try fix that!

**Comment by heada** [comment]
@trsvchn check out my last commit. Tell me If something else incorrect.

**Comment by trsvchn** [comment]
Thanks for the update! Going to try it locally!

File: cocoviewer.py [hunk 1]
```diff
@@ -1,6 +1,5 @@
 #!/usr/bin/env python3
 """COCO Dataset Viewer.
```
Comment (trsvchn) [inline cocoviewer.py:2]:
```suggestion
"""COCO Dataset Viewer.

```
Let's keep an empty line here.

File: cocoviewer.py [hunk 2]
```diff
@@ -13,7 +12,7 @@
 import tkinter.ttk as ttk
 from tkinter import filedialog, messagebox
 from turtle import __forwardmethods
-
+import numpy as np
```
Comment (trsvchn) [inline cocoviewer.py:15]:
```suggestion

import numpy as np
```
nit: add an empty line to separate imports

File: cocoviewer.py [hunk 3]
```diff
@@ -196,11 +195,26 @@ def draw_masks(draw, objects, obj_categories, ignore, alpha):
                     if m_:
                         draw.polygon(m_, outline=fill, fill=fill)
             # TODO: Fix problem with RLE
```
Comment (trsvchn) [inline cocoviewer.py:197]:
```suggestion
            # RLE mask for collection of objects (iscrowd=1)
```
nit: change comment

File: cocoviewer.py [hunk 4]
```diff
@@ -196,11 +195,26 @@ def draw_masks(draw, objects, obj_categories, ignore, alpha):
...
-            # elif isinstance(m, dict):
-            #     draw.polygon(m['counts'][1:-2], outline=c[-1], fill=fill)
+            elif isinstance(m, dict):
```
Comment (trsvchn) [inline cocoviewer.py:198]:
```suggestion
            elif isinstance(m, dict) and objects[i]["iscrowd"]:
```
According to [format-data](https://cocodataset.org/#format-data):
> The segmentation format depends on whether the instance represents a single object (iscrowd=0 in which case polygons are used) or a collection of objects (iscrowd=1 in which case RLE is used).

Let's check that here

File: cocoviewer.py [hunk 5]
```diff
@@ -196,11 +195,26 @@ def draw_masks(draw, objects, obj_categories, ignore, alpha):
...
+  rlePairs = np.array(rle).reshape(-1,2)
+  img = np.zeros(rows*cols, dtype=np.uint8)
+  index_offset = 0
+  for index, length in rlePairs:
+    index_offset += index
+    img[index_offset:index_offset+length] = 255
+    index_offset+=length
+  img = img.reshape(cols,rows)
+  img = img.T
+  return img
```
Comment (trsvchn) [inline cocoviewer.py:217]:
```suggestion

def rle_to_mask(rle, height, width):
    rows, cols = height, width
    rle_pairs = np.array(rle).reshape(-1, 2)
    img = np.zeros(rows * cols, dtype=np.uint8)
    index_offset = 0

    for index, length in rle_pairs:
        index_offset += index
        img[index_offset:index_offset + length] = 255
        index_offset += length

    img = img.reshape(cols, rows)
    img = img.T
    return img

```
Python code style: let's use common python coding styles: snake case for var names, 4 spaces for indentation, spacing after `,`, 2 empty lines between high level functions definitions.

File: cocoviewer.py [hunk 6]
```diff
@@ -798,4 +812,4 @@ def main():
 
 
 if __name__ == "__main__":
-    main()
+    main()
```
Comment (trsvchn) [inline cocoviewer.py:815]:
```suggestion
    main()

```
nit: newline at the end of a file

**Review by trsvchn — CHANGES_REQUESTED** [review]
@heada Thanks for the PR and updates, I've tried it locally and it works. I left some comments, mostly some pycode style nits + some additional check for rle.

**Comment by heada** [comment]
+ allright

**Review by heada — COMMENTED** [review]
allright

**Review by trsvchn — APPROVED** [review]
@heada LGTM! Many thanks for the PR!

**Pull request merged**